test/freenet/crypt/CTRBlockCipherTest.java

```diff
@@ -226,7 +226,7 @@ private void checkNISTRandomLength(int bits, byte[] key, byte[] iv,
 				}
 				c.doFinal(plaintext, 0, plaintext.length - inputPtr, output,
 						outputPtr);
-				assertTrue(Arrays.equals(output, ciphertext));
+				assertArrayEquals(output, ciphertext);
```

Comment:
I believe in all those assert methods the first parameter is the _expected_ one, and the second one is the actual. For the actual comparison this does not matter, but the error message will have the compared objects switched which _will_ confuse people.

However, if you’re already fiddling with the asserts, I’d much rather you’d switch them to using the Hamcrest matchers, because that will immediately remove any confusion that might possibly arise from the parameter order, like so:

```suggestion
				assertThat(output, equalTo(ciphertext));
```

(And yes, the matcher returned by `equalTo` knows how to compare arrays correctly. 🙂)

Reply:
Switched to hamcrest — thank you!